bundles/org.eclipse.ui.ide/src/org/eclipse/ui/internal/ide/ChooseWorkspaceDialog.java

```diff
@@ -476,6 +472,26 @@ public void widgetSelected(SelectionEvent e) {
 				}
 			}
 		});
+		parent.addControlListener(new ControlListener() {
+
+			@Override
+			public void controlResized(ControlEvent e) {
+				// browseButton
+				Point size = parent.getSize();
+				if (size.x < 300) {
```

Comment:
Why 300 as magic number here?

Reply:
I just tried it out and found that it at least is good for a first iteration.

Reply:
Hardcoding magic numbers is a bad practice. You should find a way to deduce a value from the button size or the stringExtent, or whatever; but not hardcode a randomish number.

Reply:
Feel free to suggest something, I have not really a better idea here and as all swt layout use "hardcoded numbers" (e.g. non of them allows percentages or other relatives like css) and have no way for really responsive layouts (see https://github.com/eclipse-platform/eclipse.platform.swt/issues/355) this is still better than nothing for now.

Reply:
You can implement something based on percentage: get the initial button width and the parent width; and decide from those of a condition in which button should be shortened, for example `if (size.x < 3 * initialButtonSize.x)`

Reply:
So what really matters is more the amount of characters that can fit in the text widget (overall 50). The size could be deduced from it, as this depends eg on the system font.

Reply:
Not really, as you can have long path, short path, ... in the end small is small (maybe too small as @merks suggested) so all we can do is to same some little space to show maybe some more characters. just from UX, it might be even better to have a text-field that works like CLabel with an elepis (...) in the middle but that far out of scope here.

The inital scope was, that on small screens you cannot use the dialog at all (whether or not Eclipse itself would be usable on such a screen). AND there was a complain that on larger screens the size is not used but all grey.

So this just improves the cases in two way:

1. define what "small" is (== 300) and try to give some more space, and shrink the box so probably users can see a bit more
2. If it is "large" ( > 300)  stretch the text-box so it fills all the space.

Not more not less, no fancy computation just a little bit more usable.